lib/help_scout/api/access_token.rb

```diff
@@ -23,29 +22,41 @@ def refresh!
         private
 
         def token_request_params
-          @token_request_params ||= {
+          @_token_request_params ||= {
             grant_type: 'client_credentials',
             client_id: HelpScout.app_id,
             client_secret: HelpScout.app_secret
           }
         end
       end
 
-      attr_reader :expires_at, :expires_in, :value
+      attr_accessor :invalid
+      attr_reader :expires_in, :value
 
       def initialize(params)
         @value = params[:access_token]
-        expires_in = params[:expires_in]
-
-        return unless expires_in
+        @expires_in = params[:expires_in]
+      end
 
-        @expires_in = expires_in
-        @expires_at = Time.now.utc + expires_in
+      def expires_at
+        @_expires_at ||= Time.now.utc + expires_in
       end
 
       def expired?
         Time.now.utc > expires_at
       end
+
+      def invalid?
+        invalid
+      end
+
+      def mark_invalid!
```

Comment:
I agree with you @jcypret -- I don't mind them _as much_ in, say `aasm`, because it can be difficult to correctly inflect statuses and `mark_#{status}` is easier for the gem to implement, but if we're hand-writing the methods I'd prefer to see `invalidate!` as well.

Reply:
Changed 👍